src/main/java/com/telefonica/iot/cygnus/backends/postgresql/PostgreSQLBackend.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2015 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-cygnus (FI-WARE project).
+ *
+ * fiware-cygnus is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ * fiware-cygnus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with fiware-cygnus. If not, see
+ * http://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License please contact with iot_support at tid dot es
+ */
+
+package com.telefonica.iot.cygnus.backends.postgresql;
+
+/**
+ * Interface for those backends implementing the persistence in PostgreSQL.
+ *
+ * @author hermanjunge
+ */
+public interface PostgreSQLBackend {
+
+    /**
+     * Available table types.
+     */
+    public enum TableType { TABLEBYDESTINATION, TABLEBYSERVICEPATH }
```

Comment:
In the meantime, we have added a general `data_model` parameter that is configured at `OrionSink` level. This new parameter may have values `dm-by-service`, `dm-by-service-path`, `dm-by-entity` and `dm-by-attribute`. `dm-by-service-path` matches `table-by-service-path`,  and `dm-by-entity` matches `table-by-destination`. Thus, now there is no need to declare this enum here, and you can use the general DataModel type.